synfig-studio/src/gui/widgets/widget_timeslider.h

```diff
@@ -75,6 +84,9 @@ class Widget_Timeslider: public Gtk::DrawingArea
 
 	virtual void draw_background(const Cairo::RefPtr<Cairo::Context> &cr);
 
+	void get_bounds_rectangle_dimensions(double& x0, double& x1, double& w , bool lower);
+	const double boundary_dimension =8;
```

Comment:
ugh. It is used in more than one method, so we cannot reduce its scope.
However, if we decide to change the default value, it will force recompilation of any file that `#include`s Widget_TimeSlider.h

@ice0  opinion?

Reply:
Agree. 
@mohamedAdhamc Can you explain how it is used?

Reply:
`boundary_dimension` is the width of the draggable handle.
 It's used in 
1- Widget_Timeslider::on_draw in the process of rendering using cairo to set the dimensions for the bound rectangles and the bound handle.
2- it is also used in the button press and motion notify event to know if the press/ position of the mouse is within the boundaries of the "bound handle"

I think then it should probably be initialized in the constructor

Reply:
I think it might be much easier to use `Gtk::Image` for handles?
It supports event handling, knows the position and size, and we can control the boundaries with margins.
Am I missing something here?

Reply:
> I think it might be much easier to use `Gtk::Image` for handles? It supports event handling, knows the position and size, and we can control the boundaries with margins. Am I missing something here?

well the thing is the whole "timetrack" is drawn with cairo surfaces, the play bounds, the patterns, etc... and in it isnt resizable. so I think it kind of makes sense to have this bound handle be the same as well to be moved in the same manner and for overall consistency. An image may make it easier to be resized and all (even though really that isnt needed the only thing needed if a user would want to replace the icon he just needs to make sure to replace it with an icon of the same size and everything should be fine), and I'm not sure how reliable it would be in terms of being moved around the time track .

Reply:
I don't know if it be better, but you could use a `Gtk::Overlay` that contains a `Gtk::Fixed` to then include the `Gtk::Image`s